aws/amazon-vpc-resource-controller-k8s
# remove workarounds from custom pod controller and refactor code

*Description of changes:*

- Removing workaround from the memory optimized Pod controller to pass Delete Event with request.
- Renaming file and structs, interfaces names to more descriptive names.
- Refactor code so it's easier to add new resource like IPv4 Address on top of Pod ENI.

**Testing for the Change**
- Ran Integration test in loop (x5 times) - Passed
- Ensure memory doesn't increase as the Number of Pods increases - Passed
- Ensure existing Pod's networking is not removed once the controller restarts - Passed

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
